bundle/bundler/src/lib.rs

```diff
@@ -0,0 +1,220 @@
+use std::collections::BTreeMap;
+use std::path::{Path, PathBuf};
+
+use async_std::channel::bounded;
+use async_std::task;
+use async_std::task::block_on;
+
+use anyhow::Context;
+use anyhow::Result;
+use cid::multihash::Code;
+use cid::Cid;
+use fvm_shared::actor;
+use fvm_shared::actor::builtin::Manifest;
+use fvm_shared::blockstore::{Block, Blockstore, MemoryBlockstore};
+use fvm_shared::encoding::DAG_CBOR;
+use ipld_car::CarHeader;
+
+const IPLD_RAW: u64 = 0x55;
+
+/// A library to bundle the Wasm bytecode of builtin actors into a CAR file.
+///
+/// The single root CID of the CAR file points to an CBOR-encoded IPLD
+/// Map<Cid, i32> where i32 is to be interpreted as an
```

Comment:
Hm. Maps generally map to "objects" in IPLD, so non-string keys can be an issue. I'd either:

1. Use string keys. This would let us, use nice names, version, etc.
2. Use a list of "object". I.e., `Actor { type: ..., version: ..., code: Cid }`.

Reply:
_Technically_ this can be fixed by, e.g., specifying that the map should be encoded as a list of tuples. But I don't think we have a way to specify that right now.

For the moment, everything will "work", but that'll stop working when we get stricter about what can and cannot go into an IPLD object.

Reply:
Hm, I see the point but I'm inclined to punt this until it becomes a problem. This bundle format is not network-dependent, so it won't be hard to change when the time comes.

> Use string keys. This would let us, use nice names, version, etc.
But we'd be incurring in the cost of reparsing every CID with every Machine construction. Makes me sad.

> Use a list of "object". I.e., Actor { type: ..., version: ..., code: Cid }.
Similar to above, we'd deserialize into a list and then transform it into a map (or perform O(n) traversals on every lookup).

We could overcome the above problems by memoizing the Manifest in a cache, but at this stage we don't need that sophistication, so it's not worth spending the time on that.

Reply:
Tracking in https://github.com/filecoin-project/builtin-actors/issues/9.